components/TARGET_PSA/services/attestation/COMPONENT_PSA_SRV_IMPL/attestation_bootloader_data.c

```diff
@@ -54,11 +55,13 @@ const uint8_t temp_ram_page_data[] = {
 /* Temporary Implementation ID data: mandatory claim represents the original
 ** implementation signer of the attestation key and identifies the contract
 ** between the report and verification */
-#define TEMP_IMPL_ID_DATA_SIZE (32u)
 
-#define TEMP_IMPL_ID_DATA  0xD0, 0xD1, 0xD2, 0xD3, 0xD4, 0xD5, 0xD6, 0xD7, \
-                           0xD8, 0xD9, 0xDA, 0xDB, 0xDC, 0xDD, 0xDE, 0xDF, \
-                           0xD0, 0xD1, 0xD2, 0xD3, 0xD4, 0xD5, 0xD6, 0xD7, \
-                           0xD8, 0xD9, 0xDA, 0xDB, 0xDC, 0xDD, 0xDE, 0xDF
+const uint8_t temp_impl_id_data[IMPL_ID_DATA_SIZE] = {  0xD0, 0xD1, 0xD2, 0xD3, 0xD4, 0xD5, 0xD6, 0xD7, 
+                                                        0xD8, 0xD9, 0xDA, 0xDB, 0xDC, 0xDD, 0xDE, 0xDF, 
+                                                        0xD0, 0xD1, 0xD2, 0xD3, 0xD4, 0xD5, 0xD6, 0xD7, 
+                                                        0xD8, 0xD9, 0xDA, 0xDB, 0xDC, 0xDD, 0xDE, 0xDF };
+
+const char temp_profile_def_data[PROFILE_DEFINITION_DATA_SIZE] = "psa-attest.md";
+const char temp_verification_service_data[VERIFICATION_SERVICE_DATA_SIZE] = "www.mbed.com";
```

Comment:
this code assumes a correlation between PROFILE_DEFINITION_DATA_SIZE and the string "psa-attest.md" and also between VERIFICATION_SERVICE_DATA_SIZE and the string "www.mbed.com". this is subject to change which could cause unexpected failures later. it should at the very least be documented (if not explicitly enforced) if we take this approach.

Reply:
I also was taking a close look at this earlier today. My conclusion is that if actual data is smaller/shorter then the size, static initializer will zero the rest. If the const data is larger then defined size, compilation error will ocure (tested on GCC).